BPA/Ch.10/js_rest_occ.stan

```diff
@@ -0,0 +1,232 @@
+// Jolly-Seber model as a restricted occupancy model
+
+functions {
+  // These functions are derived from Section 12.3 of
+  // Stan Modeling Language User's Guide and Reference Manual
+
+  /**
+   * Return a integer value of first capture occasion
+   *
+   * @param y_i Integer array of capture history
+   *
+   * @return Integer value of first capture occasion
+   */
+  int first_capture(int[] y_i) {
+    for (k in 1:size(y_i))
+      if (y_i[k])
+        return k;
+    return 0;
+  }
+
+  /**
+   * Return a integer value of last capture occasion
+   *
+   * @param y_i Integer array of capture history
+   *
+   * @return Integer value of last capture occasion
+   */
+  int last_capture(int[] y_i) {
+    for (k_rev in 0:(size(y_i) - 1)) {
+      int k;
+      k <- size(y_i) - k_rev;
+      if (y_i[k])
+        return k;
+    }
+    return 0;
+  }
+
+  /**
+   * Return a matrix of uncaptured probability
+   *
+   * @param nind        Number of individuals
+   * @param n_occasions Number of capture occasions
+   * @param p           Detection probability for each individual
+   *                    and capture occasion
+   * @param phi         Survival probability for each individual
+   *                    and capture occasion
+   *
+   * @return Uncaptured probability matrix
+   */
+  matrix prob_uncaptured(int nind, int n_occasions,
+                         matrix p, matrix phi) {
+    matrix[nind, n_occasions] chi;
+
+    for (i in 1:nind) {
+      chi[i, n_occasions] <- 1.0;
+      for (t in 1:(n_occasions - 1)) {
+        int t_curr;
+        int t_next;
+
+        t_curr <- n_occasions - t;
+        t_next <- t_curr + 1;
+        chi[i, t_curr] <- (1 - phi[i, t_curr]) +
```

Comment:
I have a very strong preference for operators beginning lines so that code's easy to scan down the left.  Ideally, each additive term gets its own line.  So that'd be:

```
chi[i, t_curr] <- (1 - phi[i, t_curr])
    + phi[i, t_curr] * (1 - p[i, t_next]) *  chi[i, t_next];
```